### ISSUE TITLE
Post formats: Archive templates have no description and use the slug instead of a readable name

### ISSUE THREAD

**Issue by carolinan** [WordPress/gutenberg#65319]
### Description

When a block theme includes the HTML file for a post format archive, the Site Editor and Data Views shows the template without a description and without a human-readable name. Instead it uses the slug, name from the HTML file.

It is as if the Data Views do not recognise that the template is part of the template hierachy.

### Step-by-step reproduction instructions

First, you need a block theme, and you need to enable post format support on the active theme. 

Example:
```
function twentytwentyfour_setup(){
	add_theme_support( 'post-formats', array( 'standard', 'aside', 'gallery', 'audio', 'video', 'link', 'image', 'chat', 'status', 'quote' ) );
}

add_action( 'after_setup_theme', 'twentytwentyfour_setup' );
```

Next, add a post format archive template inside the `templates` folder.
The content of the file does not matter for this test. I have used the link post format as example:
`templates/taxonomy-post_format-post-format-link.html`

Optionally, create one or more posts with the link post format assigned.

Go to Appearance > Editor > Templates.
Note that the link archive is available, but uses the whole slug as the template name.

### Screenshots, screen recording, code snippet

<img width="296" alt="The link post format archive template on the Template screen, showing the partial file name for the template." src="https://github.com/user-attachments/assets/16c0f1b7-0160-4dd6-972e-564c9591f3cb">

### Environment info

WordPress 6.6.2, with and without Gutenberg trunk.

### Please confirm that you have searched existing issues in the repo.

- [X] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [X] Yes

**Comment by carolinan** [comment]
TLDR: It seems the data for this template does not exist in the entity records.

**Comment by t-hamano** [comment]
The title and description of the default templates are defined in the core: https://github.com/WordPress/wordpress-develop/blob/a78540b0881a195ec8488bb6bdf878114d75f8b8/src/wp-includes/block-template-utils.php#L141

Therefore, it may be a good idea to submit a core ticket.

By the way, you can change the title and description by using the `default_template_types` filter hook:

```php
function twentytwentyfour_add_default_template( $default_template_types ) {
	$default_template_types['taxonomy-post_format-post-format-link'] = array(
		'title'       => __( 'Link Post Format' ),
		'description' => __( 'Description.' ),
	);
	return $default_template_types;
}
add_action( 'default_template_types', 'twentytwentyfour_add_default_template' );
```

This hook is not only reflected in the DataViews, but also in the template sidebar, for example:

![image](https://github.com/user-attachments/assets/8aca30e6-fffc-48f9-b52a-e8a8fd17290f)

**Comment by carolinan** [comment]
Thank you!

**Comment by carolinan** [comment]
Adding a link to the core ticket and PR meanwhile: https://core.trac.wordpress.org/ticket/62326

**Comment by carolinan** [comment]
I don't think there is an actionable task for Gutenberg, the name and description needs to be added in core, this issue was kept open for visibility.

**Comment by t-hamano** [comment]
If we also need to take into account WP 6.7/6.6, which the Gutenberg plugin supports, I think an equivalent to the core patch is also needed for the Gutenberg plugin. In other words, I think we need a hook like the one below in the `lib/compat/wordpress-6.8` directory.

<details><summary>Example code</summary>

```php
function gutenberg_add_default_template( $default_template_types ) {
	$post_formats = get_post_format_strings();
	foreach ( $post_formats as $post_format_slug => $post_format_name ) {
		$default_template_types[ 'taxonomy-post_format-post-format-' . $post_format_slug ] = array(
			'title'       => sprintf(
				/* translators: %s: Post format name. */
				_x( 'Post Format: %s', 'Template name' ),
				$post_format_name
			),
			'description' => sprintf(
				/* translators: %s: Post format name. */
				__( 'Displays the %s Post Format archive.' ),
				$post_format_name
			),
		);
	}
	return $default_template_types;
}
add_action( 'default_template_types', 'gutenberg_add_default_template' );
```

</details>
